src/Sylius/Behat/Page/Shop/Product/ShowPage.php

```diff
@@ -325,7 +334,7 @@ protected function getDefinedElements(): array
             'association' => '[data-test-product-association="%associationName%"]',
             'attributes' => '[data-test-product-attributes]',
             'average_rating' => '[data-test-average-rating]',
-            'catalog_promotion' => '#promotion_label',
+            'catalog_promotion' => '.sylius_catalog_promotion',
```

Comment:
As it is shop context, you should use twig extension to generate attribute `[data-test-...]`